zerver/tests/test_subs.py

```diff
@@ -5259,6 +5259,19 @@ def test_stream_settings_for_subscribing_other_users(self) -> None:
         do_change_stream_group_based_setting(
             private_stream, "can_add_subscribers_group", members_group, acting_user=None
         )
+        result = self.subscribe_via_post(
+            self.test_user,
+            ["private_stream"],
+            {"principals": orjson.dumps([invitee_user_id]).decode()},
+        )
+        self.assert_json_success(result)
+        do_change_stream_group_based_setting(
+            private_stream, "can_add_subscribers_group", nobody_group, acting_user=None
+        )
+
+        do_change_stream_group_based_setting(
+            private_stream, "can_administer_channel_group", members_group, acting_user=None
+        )
```

Comment:
This test might be missing code to remove the user so we're not trying to add someone who is already there?